### DISCUSSION TITLE
Data storage not correct / Photos not synchronising

### DISCUSSION THREAD

**Discussion by chleeb** [opengisch/QField#4263] [Q&A]
Hi,

we noticed two bugs in the QField cloud that might be related to each other.

First, there are problems when uploading photos to the cloud. QField said the sychronisation was successful, but in some cases only some of the photos are uploaded to the cloud. This might be an known bug as something similar had been reported [here](https://github.com/opengisch/QField/discussions/3328) and [here](https://github.com/opengisch/QField/discussions/398).

Second, the calculation of the used data storage seems to be not correct. 
For example, in this project ("id893", user "mbraun") we have 8 files that sum up to about 50 MB, but it is written that the project is using 327 MB:

![image](https://user-images.githubusercontent.com/89195127/230390979-308adfe8-e998-4aad-8cd9-e4c7723cb094.png)

In this project, several photos are missing,. So could it be that the difference in data storage (~277 MB) comes from the missing photos? So that they had actually been uploaded, but are just not shown in the table and can’t be accessed? (Note: Some photos were uploaded successfully but had been deleted from the cloud with a Python script). 

This difference between shown and actual data storage is especially a pity as storage is quite limited and we even pay for more storage.

Best,
Christoph

**Comment by BoswachterMarc** [comment]
About datastorage, are you not overlooking the backup files? Hidden behind the 3 dots...

**Reply by Bvpraagh** [reply]
![image](https://github.com/opengisch/QField/assets/104828477/30b9e932-81a3-49dd-bc05-9612591c0d92)

I am having the same issue. My storage doesn't match up to data use. If I click on three dots, there only seems to be one version of each file. There is no option to delete (Which I don't think you can do if there is only one version). Could there be hidden versions elsewhere? Is there a way to delete the photos which would make more room? Only my photos came through but the geopackages didn't.

**Comment by chleeb** [comment]
Thanks for pointing out the backup files, I indeed haven't considered them. Can I somehow control which and how often files are backuped?
For example we have orthophotos (.mbtiles) to have a basemap in remote areas. They are never changed on the tablets in QField, but I have up to 9 backup versions of exactly the same file, resulting in (I assume?) 195 instead of 19.5 MB data storage. Also photos (.jpg) are backuped in several versions. 
Thus, this data storage issue might not be a bug, but, in case it doesn't exist already somewhere, a request to add a function to control the backup system.

Best,
Christoph

**Comment by wanderingnature** [comment]
I  agree with chleeb - the information screen is confusing. Different values are presented to the user and they do not agree if there are any backups (prior versions) of files. Managing the versions is quite cumbersome since the icon for a file does not indicate if there are any prior versions stored. You have to click each ... and then choose an action to delete.  There should be some option in the QField sync plugin giving a choice whether to save any versions of a file or layer at all.

**Comment by CvTooren** [comment]
I have the same problem. I have 4 projects, each with about 200 measurement points with 3 photos each. The tablet has pushed everything to the cloud, and when I retrieve the data from the cloud to my desktop, I get the message that everything is synchronized. However, the input of all projects is not complete. In addition, I have no photos in 2 of the four projects, and about half of the photos in the other 2 projects. Help!

**Comment by CACG1022** [comment]
Hello, I would like to ask if you managed to get a solution to this problem? I'm having the same problem in QfieldCloud, only some photos appear, not all of them, and I don't know how I can solve this problem. Currently the only solution he did is to export the photos manually, the problem is that there are several devices and it becomes tedious to do this process manually

**Comment by Bvpraagh** [comment]
Hi. No I Have not been able to solve the problem. I think it must be a storage issue and perhaps there are hidden files I can’t find that are taking up space. I synced the project several times through the cloud and ended up getting all the photos across in the end but not the Geo-packages so no data points.

I am going to try to reduce the size of the photos in the attribute table and see if that makes them small which will address the size issue. Otherwise I will pay for extra storage and the plan but I would like to be confident there would be better support before I pay and it doesn’t seem like it.

Otherwise unfortunately you have to manually download with a cable attached to your device and computer. 

Seems quite a number of people are having this issue so it would be good if it gets addressed. I’d just like to  know why its happening and whether it is a data storage issue.
